Sources/SwiftParser/Modifiers.swift

```diff
@@ -167,4 +148,55 @@ extension Parser {
 
     return RawDeclModifierSyntax(name: keyword, detail: detail, arena: self.arena)
   }
+
+  mutating func parseAccessLevelModifier() -> RawDeclModifierSyntax {
+    assert(atAny([.privateKeyword, .fileprivateKeyword, .internalKeyword, .publicKeyword]))
+    let name = consumeAnyToken()
+    let details: RawDeclModifierDetailSyntax?
+    if let lparen = consume(if: .leftParen) {
+      var beforeDetail: [RawSyntax] = []
+      var detail: RawTokenSyntax? = nil
+      var afterDetail: [RawSyntax] = []
+
+      let (content, rightParen) = expect(.rightParen)
```

Comment:
I’m not really a fan of taking apart the unexpected tokens here. Fortunately, with the new recovery API, you should be able to just expect the `set` keyword followed by a `)`. Setting the precedence of the `set` keyword to `weakBracketClose` will make sure that we can skip over weak punctuators (like `,` or `:`) in the details.

```swift
let unexpectedBeforeDetail: RawUnexpectedNodesSyntax?
let detail: RawTokenSyntax
if let setHandle = self.canRecoverToContextualKeyword("set", precedence: .weakBracketClose) {
  (unexpectedBeforeDetail, detail) = self.eat(setHandle)
} else {
  unexpectedBeforeDetail = nil
  detail = RawTokenSyntax(
    missing: .contextualKeyword,
    text: arena.intern("set"),
    arena: arena
  )
}
let (unexpectedBeforeRightParen, rightParen) = self.expect(.rightParen)

details = RawDeclModifierDetailSyntax(
  leftParen: lparen,
  unexpectedBeforeDetail,
  detail: detail,
  unexpectedBeforeRightParen,
  rightParen: rightParen,
  arena: arena
)
```